src/server/page_server.cpp

```diff
@@ -244,6 +258,75 @@ page_server::connection_handler::prior_sender_sink_hook (std::string &&message)
   m_conn->push (page_to_tran_request::SEND_TO_PTS_LOG_PRIOR_LIST, std::move (message));
 }
 
+page_server::disconnect_handler::disconnect_handler ()
+  : m_terminate { false }
+{
+  m_thread = std::thread (&page_server::disconnect_handler::disconnect_loop, std::ref (*this));
+}
+
+page_server::disconnect_handler::~disconnect_handler ()
+{
+  if (m_terminate.load () && m_thread.joinable ())
+    {
+      m_queue_cv.notify_one ();
```

Comment:
Do we really need this notify? It was already called when terminate was set.

Reply:
Maybe not. It is covered indeed by the terminate notify an then by the assert in `disconnect`.